ipatests/test_integration/test_acme.py

```diff
@@ -769,29 +824,213 @@ def test_pruning_negative_options(self):
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--enable'])
+        self.master.run_command(['ipactl', 'restart'])
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--run'])
+        time.sleep(100)
+
+        # check if client cert is removed
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert 'CN=%s' % self.clients[0].hostname not in result.stdout_text
+
+    def test_prune_cert_cron(self, issue_and_expire_acme_cert):
+        """Test to prune expired certificate by cron job"""
+        if (tasks.get_pki_version(self.master)
+           < tasks.parse_version('11.3.0')):
+            raise pytest.skip("Certificate pruning is not available")
+
+        issue_and_expire_acme_cert(
+            self.master, self.clients[0], self.acme_server)
+
+        # check that the certificate issued for the client
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert 'CN=%s' % self.clients[0].hostname in result.stdout_text
+
+        # enable pruning
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--enable'])
+
+        # cron would be set to run the next minute
+        cron_minute = self.master.run_command(
+            [
+                "python3",
+                "-c",
+                (
+                    "from datetime import datetime; "
+                    "print(int(datetime.now().strftime('%M')) + 1)"
+                ),
+            ]
+        ).stdout_text.strip()
+        self.master.run_command(
+            ['ipa-acme-manage', 'pruning',
+             f'--cron={cron_minute} * * * *']
+        )
+        self.master.run_command(['ipactl', 'restart'])
+        time.sleep(150)
+
+        # check if client cert is removed
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert 'CN=%s' % self.clients[0].hostname not in result.stdout_text
+
+    def test_prune_cert_retention_unit(self, issue_and_expire_acme_cert):
+        """Test to prune expired certificate with retention unit option"""
+        if (tasks.get_pki_version(self.master)
+           < tasks.parse_version('11.3.0')):
+            raise pytest.skip("Certificate pruning is not available")
+        issue_and_expire_acme_cert(
+            self.master, self.clients[0], self.acme_server)
+
+        # check that the certificate issued for the client
+        result = self.master.run_command(['ipa', 'cert-find'])
+        assert f'CN={self.clients[0].hostname}' in result.stdout_text
+
+        # enable pruning
+        self.master.run_command(['ipa-acme-manage', 'pruning', '--enable'])
+
+        # certretention set to 5 min
+        self.master.run_command(
+            ['ipa-acme-manage', 'pruning',
+             '--certretention=5', '--certretentionunit=minute']
+        )
+        self.master.run_command(['ipactl', 'restart'])
+        # self.master.run_command(
+        #     ['ipa-acme-manage', 'pruning', '--run']
+        # )
```

Comment:
Here I think you'd want to:
- call the fixture
- enable pruning with 5 minutes
- cert-find to see that is still there
- wait 5 minutes
- manual run
- cert-find to see that is gone